pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -110,6 +129,22 @@ func (cd *clusterDeploy) doSync(cluster *v3.Cluster) error {
 }
 
 func redeployAgent(cluster *v3.Cluster, desiredAgent, desiredAuth string) bool {
+	if cluster.Status.AgentImageOverride != "" {
+		//overridden agent
+		if cluster.Status.AgentImageOverride == cluster.Spec.DesiredAgentImage {
+			return false
+		}
+		if cluster.Status.AgentImageOverride == cluster.Status.AgentImage {
+			return false
+		}
+
+		//kick a redeploy with the overriden agent
+		clearAgentImages(cluster.Name)
+		cluster.Status.AgentImage = ""
+		cluster.Spec.DesiredAgentImage = cluster.Status.AgentImageOverride
+		return true
```

Comment:
for debugging purposes, let's add log statement whenever redeploy returns true same as L168

Reply:
i agree, will do